internal/applicationsnapshot/input.go

```diff
@@ -183,15 +183,27 @@ func DetermineInputSpec(ctx context.Context, input Input) (*app.SnapshotSpec, *E
 }
 
 func readSnapshotSource(input []byte) (app.SnapshotSpec, error) {
-	var file app.SnapshotSpec
-	err := yaml.Unmarshal(input, &file)
-	if err != nil {
+	// Define a temporary struct to capture the wrapped spec
```

Comment:
```suggestion
	// Define a temporary struct to capture the wrapped spec so we
	// so can read snapshot data correctly from a cluster record
```

It's pretty clear what's going on here I guess, but maybe some extra commentary would be nice.